cookbooks/aws-parallelcluster-environment/recipes/config/network_interfaces.rb

```diff
@@ -14,8 +14,8 @@
 
 return if on_docker?
 
-def device_number(mac, token)
-  uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/device-number")
+def network_card_index(mac, token)
+  uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/network-card")
```

Comment:
Could you please add more details about the fix in the commit message?

Why do we need to use network card index rather than device number? 
Which was the issue? Which is the new behaviour?

Reply:
Done, added more description

Reply:
I randomly checked the metadata from a t2.medium, and the `network-card` field is not available there.

How can we be sure that that field is available at least on all multi-NIC instance types we support?

Reply:
The logic is only run on multi-nics instances. We have integration tests running on multiple mult-nics instances and didn't discover any issues.

Reply:
How many different multi-NIC instance types are we checking?